### PR TITLE
Implement convenience arguments for device update

### PR DESCRIPTION
**Pull request by anusapan**
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [x] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [x] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.
- [x] Have you made an entry in HISTORY.rst which concisely explains your feature or change?

### CODE REVIEW

**Comment by anusapan** [comment]
Unit Test
image

Integration Test 
image

**Comment by anusapan** [comment]
Fix for issue https://github.com/Azure/azure-cli/issues/2531

File: azext_iot/operations/hub.py [hunk 1]
```diff
@@ -246,6 +246,38 @@ def _create_self_signed_cert(subject, valid_days, output_path=None):
...
+        if auth_method == DeviceAuthType.shared_private_key.name:
+            auth = 'sas'
+            if (primary_key and not secondary_key) or (not primary_key and secondary_key):
+                raise CLIError("primary + secondary Key required with sas auth")
+            instance['authentication']['symmetricKey']['primaryKey'] = primary_key
+            instance['authentication']['symmetricKey']['secondaryKey'] = secondary_key
+        elif auth_method == DeviceAuthType.x509_thumbprint.name:
+            auth = 'selfSigned'
+            if not any([primary_thumbprint, secondary_thumbprint]):
+                raise CLIError("primary + secondary Thumbprint required with selfSigned auth")
```
Comment (digimaun) [inline azext_iot/operations/hub.py:268]: If this is doing `not any([primary_thumbprint, secondary_thumbprint])` then the error should say Primary **or** Secondary Thumbprint required with selfSigned auth. I would also double check if you can have just a secondary thumbprint. As in check if the primary thumbprint is minimum required or not.

File: azext_iot/operations/hub.py [hunk 2]
```diff
@@ -246,6 +246,38 @@ def _create_self_signed_cert(subject, valid_days, output_path=None):
...
+    if edge_enabled is not None:
+        instance['capabilities']['iotEdge'] = edge_enabled
+    if status is not None:
+        instance['status'] = status
+    if status_reason is not None:
+        instance['statusReason'] = status_reason
+    if auth_method is not None:
+        if auth_method == DeviceAuthType.shared_private_key.name:
+            auth = 'sas'
+            if (primary_key and not secondary_key) or (not primary_key and secondary_key):
```
Comment (digimaun) [inline azext_iot/operations/hub.py:261]: Does the service support providing your own SAS key for devices? From what I last recall, you are able to regenerate the primary or secondary key by setting it to an empty value. Let's validate whether you are able to provide a custom key here, and if not when the user changes from some other auth mode to shared private key, this functionality should automatically regenerate keys for them.

If custom shared private key is not possible, then we don't need the `primary_key` and `secondary_key` parameters. Instead we can have a `regenerate_keys` / `--regenerate-keys` where if `auth_method == DeviceAuthType.shared_private_key.name` this function will set both `instance['authentication']['symmetricKey']['primaryKey']=''` and `instance['authentication']['symmetricKey']['secondaryKey'] = ''` which _should_ regenerate them.

File: azext_iot/operations/hub.py (hunk 1 above)
Comment (anusapan) [reply azext_iot/operations/hub.py]: Done

File: azext_iot/operations/hub.py (hunk 2 above)
Comment (anusapan) [reply azext_iot/operations/hub.py:261]: Yes, we can provide custom keys

Comment (digimaun) [reply azext_iot/operations/hub.py:261]: OK. We should still support regenerating keys easily like in the Portal. 
image

File: azext_iot/operations/hub.py (hunk 1 above)
Comment (digimaun) [reply azext_iot/operations/hub.py]: Double checking - so you can have a secondary thumbprint but not primary? Is that right?

Comment (anusapan) [reply azext_iot/operations/hub.py]: Yes

File: azext_iot/operations/hub.py (hunk 2 above)
Comment (anusapan) [reply azext_iot/operations/hub.py:261]: Shouldn't we create a separate command for renewing the keys for the device-identity like https://docs.microsoft.com/en-us/cli/azure/iot/hub/policy?view=azure-cli-latest#az_iot_hub_policy_renew_key

Comment (digimaun) [reply azext_iot/operations/hub.py:261]: Yea thats fair.

**Review by digimaun — APPROVED** [review]

**Pull request merged**